shopping-cart-room/src/main/java/com/handstandsam/shoppingapp/cart/ShoppingCartRoom.kt

```diff
@@ -0,0 +1,97 @@
+package com.handstandsam.shoppingapp.cart
+
+import android.content.Context
+import androidx.room.Room
+import com.handstandsam.shoppingapp.models.Item
+import com.handstandsam.shoppingapp.models.ItemWithQuantity
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.launch
+
+
+class ShoppingCartRoom(appContext: Context) :
+    CoroutineScope by CoroutineScope(Dispatchers.IO),
+    ShoppingCart {
+
+    private val itemInCartDatabase: ItemInCartDatabase = Room.databaseBuilder(
+        appContext,
+        ItemInCartDatabase::class.java,
+        "cart_room"
+    ).build()
+
+    private val itemInCartDao: ItemInCartDao = itemInCartDatabase.itemInCartDao()
+
+    private val channel = ConflatedBroadcastChannel(listOf<ItemWithQuantity>())
+
+    init {
+        itemInCartDao.selectAllStream().observeForever { list ->
+            launch {
+                channel.send(list.toItemWithQuantityList())
+            }
+        }
+    }
+
+    override suspend fun empty() {
```

Comment:
I haven't pulled this on my computer, but I expect that because `itemInCartDao.empty()` is not a suspending function, then the suspend in this method will be greyed out and the IDE will tell you that the suspend modifier isn't needed. 
You could just take advantage of the coroutines support in room and make the `itemInCartDao.empty()` suspending. Same for all the other synchronous methods in Room.

Reply:
It turns out I really don't need the `suspend` on the Room database because I've wrapped all the calls with my own suspending functions